edgedb.go

```diff
@@ -33,8 +34,53 @@ import (
 	"github.com/edgedb/edgedb-go/internal/soc"
 )
 
+const defaultMaxTxRetries = 3
+
 var rnd = rand.New(rand.NewSource(time.Now().UnixNano()))
 
+func defaultBackoff(attempt int) time.Duration {
+	backoff := math.Pow(2.0, float64(attempt)) * 100.0
+	jitter := rnd.Float64() * 100.0
+	return time.Duration(backoff+jitter) * time.Millisecond
+}
+
+// Action is work to be done in a transaction.
+type Action func(context.Context, Tx) error
+
+// Trier allows trying actions in a transaction.
+type Trier interface {
+	// TryTx runs an action in a transaction.
+	// If the action returns an error the transaction is rolled back,
+	// otherwise it is committed.
+	TryTx(context.Context, Action) error
+
+	// Retry does the same as TryTx
+	// but retries the action if it might succeed on a subsequent attempt.
+	Retry(context.Context, Action) error
+}
+
+// Querier allows querying the database.
+type Querier interface {
```

Comment:
I used this name because it seemed to follow [this convention](https://golang.org/doc/effective_go.html#interface-names), but I'm happy to change it to `Executor`.

Reply:
changed